### DISCUSSION TITLE
A workaround for bidirectional/inout pins

### DISCUSSION THREAD

**Discussion by eddiewastaken** [logisim-evolution/logisim-evolution#792] [Show and tell]
Here's an implemention I made of the [74LS245 (Bus Transceivers)](https://www.ti.com/lit/ds/symlink/sn54ls245-sp.pdf). The direction control makes it easy to reverse the direction without conflicting signals. Obviously in AND out pins had to be used, but this is the neatest solution I could manage for now:

![bidirect](https://user-images.githubusercontent.com/45104634/74149241-39ff5580-4bff-11ea-82b5-f938d5846a0b.png)

Hopefully this helps someone!

**Comment by ReGuess** [comment]
A slight modification, making use of components' variable bit-width:

![bidirectional](https://user-images.githubusercontent.com/7409957/82278517-f2186900-9979-11ea-82c3-3d752f99e829.png)
[bidirectional.zip](https://github.com/reds-heig/logisim-evolution/files/4647935/bidirectional.zip)

**Comment by TChapman500** [comment]
You only need one group of tri-state buffers.

**Comment by maehne** [comment]
Support for bidirectional pins is tracked in issue #253. I convert this issue describing possible workarounds to a discussion.
